pool/src/transaction_pool.rs

```diff
@@ -145,8 +151,12 @@ impl TransactionPool {
 		self.is_acceptable(&tx, stem)?;
 
 		// Make sure the transaction is valid before anything else.
-		tx.validate(self.verifier_cache.clone())
-			.map_err(PoolError::InvalidTx)?;
+		// Validate tx accounting for max tx weight.
+		tx.validate(
```

Comment:
But when adding a tx to the pool we _also_ want to validate the tx "in isolation" and when we do this we do want to verify it is below the max weight limit.

So we're using tx validation for two different purposes here - 
* is the tx itself valid
* does it produce a valid aggregate tx when added to the txpool (might be _huge_)